src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Sets a build time in minutes <see cref="Goal"/> for a given Definition in a Channel. This is captured for the Power BI Dashboard -Internal Report under .Net Core Engineering Services workspace.
+        /// </summary>
+        /// <param name="goalData">An object containing build time goal in minutes <see cref="Goal"/></param>
+        /// <param name="channelName">Channel Name for the build time Eg. .Net Core 5</param>
+        /// <param name="definitionId">Azure DevOps pipeline Definition Id</param>
+        [HttpPut]
```

Comment:
There is no router here... where are "channelName" and "definitionId" supposed to come from?

Reply:
I was committing this change when you commented.  It's updated now :)